### DISCUSSION TITLE
REVEN does not delete replay files

### DISCUSSION THREAD

**Discussion by johdcmlaker2000** [tetrane/reven-community#8] [Q&A] [answered]
It seems that REVEN does not remove the replay files after they has been deleted from Scenario list.
I deleted replay file and REVEN VM machine still takes a lot disk space (the same as a replay file size was).
Ho to fix this? Or how can i delete replay files manually, where they stored?

**Comment by Hyask** [comment, marked as answer]
Hi,

Such issue is not known and deleting the scenario from the scenario deletes its replayed resources when tested locally.

Could you provide us with additional information on your setup? It looks like you installed REVEN in a VM? If so, using which hypervisor? Which disk usage does the VM itself reports (e.g. if you run `df -h` from inside your REVEN VM)?

A possible issue would be the virtual disk of the REVEN VM growing to give enough space for the scenario files, and then not shrinking on its own when the space becomes available again.

If so, the way to shrink the virtual disk again is specific to your hypervisor. For instance here's such a [procedure for Virtualbox](https://www.howtogeek.com/312883/how-to-shrink-a-virtualbox-virtual-machine-and-free-up-disk-space/).

**Reply by johdcmlaker2000** [reply]
Hyask, thanks for reply. Seems that this is some hypervisor image caching problem. 

test@debian:~$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev             16G     0   16G   0% /dev
tmpfs           3.2G  9.4M  3.2G   1% /run
/dev/sda1       1.3T  174G  1.1T  15% /
tmpfs            16G   16K   16G   1% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
tmpfs            16G     0   16G   0% /sys/fs/cgroup
tmpfs           3.2G  3.4M  3.2G   1% /run/user/1000
/dev/sr0        3.7G  3.7G     0 100% /media/cdrom0

I make a "Compact disk" VMWare process for selected image but nothing has changed, the size of the REVEN image on the disk remains unchanged (1.3T).

**Reply by Hyask** [reply]
Hi,

That's a common problem with virtual disks. Dynamic resizing make them grow easily, but it's always a pain to shrink
them. I don't know much about VMWare, but here are some pointers:
* Did you zerofill the disk before using the "Compact disk" option? That's done by running something like the following on the guest as root:
  `dd if=/dev/zero of=/zerofill; rm /zerofill`
  Once the command completes, you will have filled your virtual disk with zeros, and the "Compact disk" option may be more successful in its shrinking process.  
  Sadly, this will also maximize the disk size before the shrink.
* If the "Compact disk" still doesn't work, I know that using `qemu-img convert` on disk images is usually a good way to make a smaller disk from a big one. This will give you a second disk image that you can just drop in place of the first one while the VM is powered off.  
  After a zerofill performed as described above, usage is as follow to make a new `vmdk` image:
  `qemu-img convert -O vmdk path/to/original/big/disk.vmdk path/to/new/smaller/disk.vmdk`
* A possible way of avoiding that could be to use a shared folder (or a network storage) with the VM to store the Reven data (the `QUASAR_ROOT` configuration option documented [here](https://doc.tetrane.com/free/latest/Installation/Prerequisites/System.html#storage)). That way the VM disk size will only ever contain the system and shouldn't grow that much, while the scenario data will live on the host's filesystem that should handle deletion correctly.  
  A downside to this will probably be performances, on which I can provide little to none information as each setup is different.

Hope this helps! Don't hesitate to come back if you need further assistance.

**Reply by johdcmlaker2000** [reply]
Thanks, Hyask

"dd if=/dev/zero of=/zerofill; rm /zerofill" and Compact VM image helps in my case

**Reply by Hyask** [reply]
Great, thanks for coming back to confirm the procedure :-)
